### PR TITLE
Unit tests for nomic client and edge case fixes

### PR DESCRIPTION
**Pull request by AndriyMulyar**
Running `make test` after being logged in will run a suite of unit tests. These tests are not yet automated on circleci but do cover several edge cases we have hit in prod users putting in data.

### CODE REVIEW

File: nomic/atlas.py [hunk 1]
```diff
@@ -239,6 +240,61 @@ def _get_index_job(self, job_id: str):
...
+
+                if project['modality'] == 'text':
+                    if isinstance(datum[key], str) and len(datum[key]) == 0:
+                        msg = 'Datum {} had an empty string for key: {}'.format(datum, key)
+                        raise ValueError(msg)
+
+                if not isinstance(datum[key], (str, float, int)):
+                    raise Exception(f"Metadata sent to Atlas must be a flat dictionary. Values must be strings, floats or ints. Key `{key}` of datum {str(datum)} is in violation.")
+
+        if added_id_field_for_user:
```
Comment (bstadt) [inline nomic/atlas.py:292]: this might get loud

File: nomic/atlas.py [hunk 2]
```diff
@@ -271,6 +327,14 @@ def add_embeddings(
             raise ValueError(msg)
 
         progressive = len(project['atlas_indices']) > 0
+        try:
+            self._validate_user_supplied_metadata(data=data, project=project)
+        except BaseException as e:
+            if not progressive:
+                self.delete_project(project_id=project_id)
```
Comment (bstadt) [inline nomic/atlas.py:334]: not sure we want to do this here. may be confusing if a user is not going through the map_embeddings interface

**Review by bstadt — APPROVED** [review]
